src/api/js/scripts/parse-api.js

```diff
@@ -0,0 +1,354 @@
+'use strict';
+
+let fs = require('fs');
+let path = require('path');
+
+let files = [
+  'z3_api.h',
+  'z3_algebraic.h',
+  'z3_ast_containers.h',
+  'z3_fixedpoint.h',
+  'z3_fpa.h',
+  'z3_optimization.h',
+  'z3_polynomial.h',
+  'z3_rcf.h',
+  'z3_spacer.h',
+];
+
+let aliases = {
+  __proto__: null,
+  Z3_bool: 'boolean',
+  Z3_string: 'string',
+  bool: 'boolean',
+  signed: 'int',
+};
+
+let primitiveTypes = {
+  __proto__: null,
+  Z3_char_ptr: 'string',
+  unsigned: 'number',
+  int: 'number',
+  uint64_t: 'bigint',
+  int64_t: 'bigint',
+  double: 'number',
+  float: 'number',
+};
+
+let optTypes = {
+  __proto__: null,
+
+  Z3_sort_opt: 'Z3_sort',
+  Z3_ast_opt: 'Z3_ast',
+  Z3_func_interp_opt: 'Z3_func_interp',
+};
+
+// parse type declarations
+let types = {
+  __proto__: null,
+
+  // these are function types I can't be bothered to parse
+  Z3_error_handler: 'Z3_error_handler',
+  Z3_push_eh: 'Z3_push_eh',
+  Z3_pop_eh: 'Z3_pop_eh',
+  Z3_fresh_eh: 'Z3_fresh_eh',
+  Z3_fixed_eh: 'Z3_fixed_eh',
+  Z3_eq_eh: 'Z3_eq_eh',
+  Z3_final_eh: 'Z3_final_eh',
+  Z3_created_eh: 'Z3_created_eh',
+};
+
+let defApis = Object.create(null);
+let functions = [];
+let enums = Object.create(null);
+for (let file of files) {
+  let contents = fs.readFileSync(path.join(__dirname, '..', '..', file), 'utf8');
+
+  // we _could_ use an actual C++ parser, which accounted for macros and everything
+  // but that's super painful
+  // and the files are regular enough that we can get away without it
+
+  // we could also do this by modifying the `update_api.py` script
```

Comment:
I don't see adding things to update_api as a goal. It doesn't serve a purpose in itself. At best one can reuse some feature from update_api, but this seems needless now.
The goals are to ensure the bindings can be built on every push and the bindings can be exported (to npm).